js/postDetail.js

```diff
@@ -0,0 +1,54 @@
+import { getAuthHeader } from './auth.js';
+import { API_BASE_URL } from '../js/constants.js';
+/**
+ * Event listener for DOMContentLoaded to fetch post by ID from URL parameters.
+ * @listens document:DOMContentLoaded
+ */
+document.addEventListener('DOMContentLoaded', () => {
+    const urlParams = new URLSearchParams(window.location.search);
+    const postId = urlParams.get('id');
+    if (postId) {
```

Comment:
What happens when this return false? One thing you could do here is to send a message informing about what has happened

Reply:
Thanks coach, with these changes, when a user visits the page without a valid postId, the Bootstrap modal will pop up, displaying the error message. The user can then close the modal by clicking the "Close" button or anywhere outside the modal.